website/src/components/Chat/ChatAssistantDraftViewer.tsx

```diff
@@ -0,0 +1,234 @@
+import {
+  Avatar,
+  Box,
+  Button,
+  CircularProgress,
+  Collapse,
+  Divider,
+  Flex,
+  Tag,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { lazy, useEffect, useState, useCallback } from "react";
+import { ChevronLeft, ChevronRight, RotateCcw } from "lucide-react";
+import { useTranslation } from "next-i18next";
+
+import { BaseMessageEmojiButton } from "../Messages/MessageEmojiButton";
+import { InferenceMessage } from "src/types/Chat";
+import { MessageInlineEmojiRow } from "../Messages/MessageInlineEmojiRow";
+import { PluginUsageDetails } from "../Messages/PluginUsageDetails";
+import { WorkParametersDisplay } from "./WorkParameters";
+const RenderedMarkdown = lazy(() => import("../Messages/RenderedMarkdown"));
+
+type OnDraftPickedFn = (chat_id: string, regen_index: number, message_index: number) => void;
+type OnRetryFn = (params: { parentId: string; chatId: string }) => void;
+
+type ChatAssistantDraftViewerProps = {
+  chatId: string;
+  streamedDrafts: string[];
+  draftMessages: InferenceMessage[][];
+  onDraftPicked: OnDraftPickedFn;
+  onRetry: OnRetryFn;
+};
+
+export const ChatAssistantDraftViewer = ({
```

Comment:
This component is way too big, we should extract it to 2 component. One for render the content, and one for display the paging logic.